src/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs

```diff
@@ -69,8 +69,7 @@ internal void BeginReadLine()
             if (_sb == null)
             {
                 _sb = new StringBuilder(DefaultBufferSize);
-                _readToBufferTask = Task.Factory.StartNew(s => ((AsyncStreamReader)s).ReadBuffer(),
-                    this, CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
+                _readToBufferTask = Task.Run((Func<Task>)ReadBuffer);
```

Comment:
:bulb: Consider renaming private method `ReadBuffer` to `ReadBufferAsync` to make it even more clear what's going on.